gulpfile.js

```diff
@@ -221,10 +223,11 @@ function dist_src() {
         '!./src/css/opensans_webfontkit/*.{txt,html}',
         '!./src/support/**'
     ];
-
+    fs.writeFileSync(DIST_DIR + 'package.json', JSON.stringify(pkg));
```

Comment:
dist_locale is done earlier, so that `dist` exists for this line.

Reply:
There's a better (or rather: more gulp-like) way to do this:

```
	const source = require('vinyl-source-stream');
    const stream = require('stream');

[...]

    var packageJson = new stream.Readable;
    packageJson.push(JSON.stringify(pkg));
    packageJson.push(null);

    return packageJson
        .pipe(source('package.json'))
		.pipe [...]
```

Reply:
Looks like that isn't quite right. I'll add the comment above so that nobody reorders.

[19:08:56] Error: File not found with singular glob: /Web/betaflight-configurator/dist/package.json (if this was purposeful, use `allowEmpty` option)
    at Glob.<anonymous> (/Web/betaflight-configurator/node_modules/glob-stream/readable.js:84:17)
    at Object.onceWrapper (events.js:315:30)
    at emitOne (events.js:116:13)
    at Glob.emit (events.js:211:7)
    at Glob._finish (/Web/betaflight-configurator/node_modules/glob/glob.js:197:8)
    at done (/Web/betaflight-configurator/node_modules/glob/glob.js:182:14)
    at Glob._processSimple2 (/Web/betaflight-configurator/node_modules/glob/glob.js:688:12)
    at /Web/betaflight-configurator/node_modules/glob/glob.js:676:10
    at Glob._stat2 (/Web/betaflight-configurator/node_modules/glob/glob.js:772:12)
    at lstatcb_ (/Web/betaflight-configurator/node_modules/glob/glob.js:764:12)

Reply:
Heh, forgot to mention that you have to remove the line

```
.pipe(gulp.src(DIST_DIR + 'package.json', { passthrougth: true }))
```

as well - then it will work:

```
mikeller@pc:~/git/betaflight-configurator$ git diff
diff --git a/gulpfile.js b/gulpfile.js
index f27179e6..83c0f415 100644
--- a/gulpfile.js
+++ b/gulpfile.js
@@ -25,6 +25,8 @@ const yarn = require("gulp-yarn");
 const rename = require('gulp-rename');
 const os = require('os');
 const git = require('gulp-git');
+const source = require('vinyl-source-stream');
+const stream = require('stream');
 
 const DIST_DIR = './dist/';
 const APPS_DIR = './apps/';
@@ -223,10 +225,15 @@ function dist_src() {
         '!./src/css/opensans_webfontkit/*.{txt,html}',
         '!./src/support/**'
     ];
-    fs.writeFileSync(DIST_DIR + 'package.json', JSON.stringify(pkg));
-    return gulp.src(distSources, { base: 'src' })
+    var packageJson = new stream.Readable;
+    packageJson.push(JSON.stringify(pkg));
+    packageJson.push(null);
+
+    return packageJson
+        .pipe(source('package.json'))
+        .pipe(gulp.src(distSources, { base: 'src' }))
         .pipe(gulp.src('manifest.json', { passthrougth: true }))
-        .pipe(gulp.src(DIST_DIR + 'package.json', { passthrougth: true }))
         .pipe(gulp.src('yarn.lock', { passthrougth: true }))
         .pipe(gulp.src('changelog.html', { passthrougth: true }))
         .pipe(gulp.dest(DIST_DIR))
mikeller@pc:~/git/betaflight-configurator$ 
```

Reply:
function getChangesetId already writes directly to the filesystem, so I wasn't worried about breaking the universe. If you say that works I'll give it a go later.

Reply:
Good point. That one could probably also be improved - I think I added that one at some point as a clutch to get finer grained versioning for debugging problems, but never got around to making it be nice.